crates/node/api/src/node.rs

```diff
@@ -102,7 +102,7 @@ pub trait BuilderProvider<N: FullNodeComponents>: Send {
     fn builder() -> Box<dyn for<'a> Fn(Self::Ctx<'a>) -> Self + Send>;
 }
 
-impl<N: FullNodeComponents> BuilderProvider<N> for () {
+impl<N> BuilderProvider<N> for () {
     type Ctx<'a> = ();
 
     fn builder() -> Box<dyn for<'a> Fn(Self::Ctx<'a>) -> Self + Send> {
```

Comment:
the motivation of the referenced pr was to make this stateful so we can move have something like an rpc builder, e.g. op rpc setup needs external input (sequencer URL)
having the NodeAddons stateful would be first step towards that.

Reply:
the one doesn't exclude the other, the node add ons can be passed to the `BuilderProvider::Ctx`